includes/Generator/OrderAttribution.php

```diff
@@ -0,0 +1,321 @@
+<?php
+/**
+ * Order Attribution data helper.
+ *
+ * @package SmoothGenerator\Classes
+ */
+
+namespace WC\SmoothGenerator\Generator;
+
+/**
+ * Order Attribution data helper class.
+ */
+class OrderAttribution {
+
+	/**
+	 * Generate order attribution data.
+	 *
+	 * @param \WC_Order $order Order.
+	 * @param array     $assoc_args Arguments passed via the CLI for additional customization.
+	 */
+	public static function add_order_attribution_meta( $order, $assoc_args = array() ) {
+
+		if ( isset( $assoc_args['skip-order-attribution'] ) ) {
+			return;
+		}
+
+		$device_type    = self::get_random_device_type();
+		$source         = 'woo.com';
+		$source_type    = self::get_source_type();
+		$origin         = self::get_origin( $source_type, $source );
+		$order_products = $order->get_items();
+		$product_url    = get_permalink( $order_products[ array_rand( $order_products ) ]->get_id() );
+		$utm_content    = [ '/', 'campaign_a', 'campaign_b' ];
+		$utm_content    = $utm_content[ array_rand( $utm_content ) ];
+
+		$meta = array();
+
+		// If the source type is admin, we only need to set the source type.
+		if ( 'admin' === $source_type ) {
+			$meta = array(
+				'_wc_order_attribution_source_type' => $source_type,
+			);
+		} else {
+			$meta = array(
+				'_wc_order_attribution_origin'             => $origin,
+				'_wc_order_attribution_device_type'        => $device_type,
+				'_wc_order_attribution_user_agent'         => self::get_random_user_agent_for_device( $device_type ),
+				'_wc_order_attribution_session_count'      => wp_rand( 1, 10 ),
+				'_wc_order_attribution_session_pages'      => wp_rand( 1, 10 ),
+				'_wc_order_attribution_session_start_time' => self::get_random_session_start_time( $order ),
+				'_wc_order_attribution_session_entry'      => $product_url,
+				'_wc_order_attribution_utm_content'        => $utm_content,
+				'_wc_order_attribution_utm_source'         => self::get_source( $source_type ),
+				'_wc_order_attribution_referrer'           => self::get_referrer( $source_type ),
+				'_wc_order_attribution_source_type'        => $source_type,
+			);
+		}
+
+		// If the source type is not typein ( Direct ), set a random utm medium.
+		if ( 'typein' !== $source_type ) {
```

Comment:
These don't need medium either
```suggestion
		if ( ! in_array( $source_type, [ 'typein', 'admin', 'mobile_app', 'unknown' ], true ) ) {
```